### PR TITLE
Added some missing multi-dm integration tests

### PR DESCRIPTION
**Pull request by quinton-hoole**
I started adding some missing multidm integration tests to confirm the results in:

https://docs.google.com/spreadsheets/d/1O2flMIe2BWo8x6N87WInqNyvGamk-F4VtV8BngATIRE/edit#gid=0

Before adding more (i.e. all of the rest of the above table), the integration tests need to be rewritten to be automatically generated and table-driven.  

I'll do that next, but in the mean time we have a few addional tests. 

I also started alphabetising the tests to make it easier to figure out what's there and what's missing.

All tests pass.

See attached test results.

[TEST-amino.run.multidm.MultiDMTestCases.xml.zip](https://github.com/amino-os/Amino.Run/files/3323598/TEST-amino.run.multidm.MultiDMTestCases.xml.zip)

### CODE REVIEW

**Comment by AmitRoushan** [comment]
@quinton-hoole 
Following test cases are removed but not added in alphabetical rearrangement 
```
testAtLeastOnceRPCConsensusRSMDHT
testAtLeastOnceRPCDHTLoadBalancedMasterSlaveSync
testAtLeastOnceRPCCacheLease
testAtLeastOnceRPCConsensusRSM
testAtLeastOnceRPCConsensusRSMDHT
testAtLeastOnceRPCTwoPCCoordinator
testAtLeastOnceRPCLockingTransaction
```
Any reason ?

**Comment by quinton-hoole** [comment]
No, that was not my intention. I'll self-review and correct as necessary
tomorrow.  Thanks for the prompt review.

File: core/src/integrationTest/java/amino/run/multidm/MultiDMTestCases.java [hunk 1]
```diff
@@ -173,143 +173,140 @@ private MicroServiceSpec createMultiDMTestSpec(String... names) throws Exception
...
-        }
-        if (name.equals("ConsensusRSM")) {
-            return "amino.run.policy.replication";
-        }
         if (name.equals("AtLeastOnceRPC")) {
             return "amino.run.policy.atleastoncerpc";
         }
         if (name.equals("CacheLease")) {
             return "amino.run.policy.cache";
         }
```
Comment (maheshrajus) [inline core/src/integrationTest/java/amino/run/multidm/MultiDMTestCases.java]: Many 'if' conditions. Can we add 'switch' case instead of 'if' condition?

Comment (quinton-hoole) [reply core/src/integrationTest/java/amino/run/multidm/MultiDMTestCases.java]: This all needs to be rewritten.  It's a mess.  In the mean time I just followed what is there.

**Comment by quinton-hoole** [comment]
Addressed all review comments, and did an additional self-review.   All tests pass.  Merging.

**Pull request merged**